tests/pthread/test_pthread_detach.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <thread>
+#include <math.h>
+#include <emscripten.h>
+#include <assert.h>
+
+#ifndef REPORT_RESULT
+#include <iostream>
+#endif
+
+extern "C" {
+//Create a thread that does some work
+void EMSCRIPTEN_KEEPALIVE spawn_a_thread() {
+	std::thread( [] {
+		double d=0;
+		for (int i=0; i<10000000; i++)			//simulate work
+			d += (i%2 ? sqrt((int)(rand())) : (-1)*sqrt((int)(rand())));
```

Comment:
For the purposes of this test, each thread can return immediately without burning CPU overhead - this is because there is a element of timing in play in this test, i.e. if this happens to take a long time or stall in the browser due to some unspecified reason, then potentially all threads might live together.

Reply:
Yeah, I guess I was just concerned that spawning a thread that does nothing could be removed by the compiler as an optimisation. And I guess I thought there might be some sort of special handling, or timing issue when trying to observe a thread that terminates instantly.
But you're right, this is probably not the best approach. I'll take another crack at writing a decent test.